hazelcast/src/main/java/com/hazelcast/map/impl/proxy/NearCachedMapProxyImpl.java

```diff
@@ -262,6 +262,12 @@ protected Data removeInternal(Data key) {
     }
 
     @Override
+    protected void removeAllInternal(Predicate predicate) {
+        super.removeAllInternal(predicate);
+        nearCache.clear();
```

Comment:
yes it can be no problem with it, hd near-cache has a different imp then hd imap